src/Microsoft.ML.Data/Evaluators/AnomalyDetectionEvaluator.cs

```diff
@@ -630,13 +630,13 @@ private protected override void PrintFoldResultsCore(IChannel ch, Dictionary<str
             {
                 int index;
                 if (!top.Schema.TryGetColumnIndex(AnomalyDetectionEvaluator.TopKResultsColumns.Instance, out index))
-                    throw Host.Except("Data view does not contain the 'Instance' column");
+                    throw Host.ExceptSchemaMismatch(nameof(top.Schema), "instance", AnomalyDetectionEvaluator.TopKResultsColumns.Instance);
                 var instanceGetter = cursor.GetGetter<ReadOnlyMemory<char>>(index);
                 if (!top.Schema.TryGetColumnIndex(AnomalyDetectionEvaluator.TopKResultsColumns.AnomalyScore, out index))
-                    throw Host.Except("Data view does not contain the 'Anomaly Score' column");
```

Comment:
>Anomaly Score' [](start = 71, length = 14)

some people would get picky about the capitalization :)

Reply:
I don't think it changes to lowercase.

However, the second argument of `ExceptSchemaMismatch` is `string columnRole`  and the message becomes:
                `return $"Could not find {columnRole} column '{columnName}'";`

So columnRole only serves the purpose of qualifying "column", while the parameter `columnName` will give the exact name. That's why I have made it lowercase everywhere.



---
In reply to: [250474268](https://github.com/dotnet/machinelearning/pull/2217#discussion_r250474268) [](ancestors = 250474268)

Reply:
The name of the column does not include a space, so the current parameter `columnRole` is already not matching the actual column name. As you see in my version, I include both the parameter `column role` which is a description of the role of the column "anomaly score", and the actual column name `AnomalyDetectionEvaluator.TopKResultsColumns.AnomalyScore` as part of the exception message. 

---
In reply to: [251125939](https://github.com/dotnet/machinelearning/pull/2217#discussion_r251125939) [](ancestors = 251125939,250823347,250474268)

Reply:
Right, this is meant to be human readable. What @Zruty0 was doing was trying to unify the error messages, I believe. And they were things like "the label column named 'Bubba' should be like this" or "the score column named 'Bizblam' should be like that," and whatnot. So I think the lowercase names for these roles specifically is appropriate, since it is just meant to be a textual message read and understood by a human.

For that same reason, I'm not too worried about these little strings everywhere. Considering that we have hundreds, possibly thousands of hand-written error messages, this seems fairly minor. So I'm not sure the situation requires something as elaborate as yet-another-static-class-full-of-strings. (I feel differently about something like `MetadataUtils.Kinds` where correctness of algorithms is at stake if we get those wrong.) At a certain point these things lend less clarity, not more, I think?